src/components/LayoutTree.tsx

```diff
@@ -454,6 +468,13 @@ const LayoutTree = () => {
     ),
   )
 
+  const { footerHeight, navbarHeight } = useNavAndFooterHeight()
+  const navAndFooterHeight = navbarHeight + footerHeight
+
+  const maxVisibleY = viewportHeight + scrollTop - ((distanceFromTop.current ?? 0) + navbarHeight)
```

Comment:
Similar to `toolbarHeightRef`, we can simplify the type of `distanceFromTop` to only be `number` rather than `number | undefined`. It's nice to handle edge cases (the element not being found) upfront rather than letting those unknown/undefined values affect downstream usage as we see here with the `?? 0` default.